vaas/vaas/adminext/widgets.py

```diff
@@ -133,3 +134,42 @@ def split_condition(value):
                 right = right[:-1]
             return left, operator, right
     return ['req.url', '~', '']
+class ComplexRedirectConditionWidget(forms.MultiWidget):
+    def __init__(self,http_methods, domains, attrs=None): 
+        widgets = ( 
+            forms.Select(choices=http_methods, attrs={'class': 'form-control','style': 'display: inline-block; width:15%'}),
+            forms.Select(choices=domains, attrs={'class': 'form-control','style': 'display: inline-block; width:40%'}),
+            forms.TextInput(attrs={'class': 'form-control','style': 'display: inline-block; width:45%','placeholder': 'Source path'}),
+        )
+        super().__init__(widgets, attrs)
+
+    def decompress(self, value):
+        if value:
+            parts = value.split(' ')
+            condition_domain = self.attrs.get('condition_domain', None)
+            if len(parts) > 1:
+                http_method = parts[2]
+                src_path = parts[6]
+                return http_method[1:-1], condition_domain, src_path[1:-1]
+        return ['GET','','']
+
+class ComplexRedirectConditionField(forms.MultiValueField):
+    def __init__(self, **kwargs):
+        configuration = provide_redirect_configuration() 
+        http_methods=tuple((http_method.http_method, http_method.name) for http_method in configuration.http_methods)
```

Comment:
It necessary to make it tuple? Because you can left statement only in curly bracers. It will return generator which is also iterable.

Reply:
It doesn't work without `tuple`